eras/babbage/impl/src/Cardano/Ledger/Babbage/Tx.hs

```diff
@@ -1,8 +1,195 @@
+{-# LANGUAGE BangPatterns #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE TypeApplications #-}
+{-# LANGUAGE TypeFamilies #-}
+{-# OPTIONS_GHC -Wno-orphans #-}
+
 module Cardano.Ledger.Babbage.Tx
-  ( module X,
-    TxBody (..),
+  ( AlonzoTx (..),
+    BabbageTxBody (..),
+    module X,
+    getDatumBabbage,
+    babbageTxScripts,
+    refScripts,
+    babbageInputDataHashes,
   )
 where
 
-import Cardano.Ledger.Alonzo.Tx as X hiding (TxBody (..))
-import Cardano.Ledger.Babbage.TxBody (TxBody (..))
+import Cardano.Ledger.Alonzo.PlutusScriptApi (getSpendingTxIn)
+import Cardano.Ledger.Alonzo.Tx as X hiding (AlonzoTxBody (..), TxBody)
+import Cardano.Ledger.Alonzo.TxSeq
+  ( AlonzoTxSeq (AlonzoTxSeq, txSeqTxns),
+    hashAlonzoTxSeq,
+  )
+import Cardano.Ledger.Alonzo.TxWitness
+  ( AlonzoEraWitnesses (..),
+    TxWitness (..),
+    addrAlonzoWitsL,
+    bootAddrAlonzoWitsL,
+    datsAlonzoWitsL,
+    rdmrsAlonzoWitsL,
+    scriptAlonzoWitsL,
+    unTxDats,
+  )
+import Cardano.Ledger.Babbage.Era (BabbageEra)
+import Cardano.Ledger.Babbage.TxBody
+  ( BabbageEraTxBody (..),
+    BabbageEraTxOut (..),
+    BabbageTxBody (..),
+    BabbageTxOut (..),
+    Datum (..),
+    dataHashTxOutL,
+  )
+import Cardano.Ledger.Core
+import qualified Cardano.Ledger.Crypto as CC
+import Cardano.Ledger.Shelley.UTxO (UTxO (..))
+import Cardano.Ledger.TxIn (TxIn)
+import Control.Applicative ((<|>))
+import Control.SetAlgebra (eval, (◁))
+import qualified Data.Map.Strict as Map
+import Data.Maybe.Strict (StrictMaybe (SJust, SNothing), strictMaybeToMaybe)
+import Data.Set (Set)
+import qualified Data.Set as Set
+import Lens.Micro
+
+instance CC.Crypto c => EraTx (BabbageEra c) where
+  type Tx (BabbageEra c) = AlonzoTx (BabbageEra c)
+  mkBasicTx = mkBasicAlonzoTx
+  bodyTxL = bodyAlonzoTxL
+  witsTxL = witsAlonzoTxL
+  auxDataTxL = auxDataAlonzoTxL
+  sizeTxG = sizeAlonzoTxG
```

Comment:
this is another one of those "getters" that we talk about today that are not actually getters. I'd love a convention that makes this clearer. We have `L` for lens, `G` for getter... maybe `F` for function?

Reply:
Yep. That sounds good. I'll rename those right now

Reply:
Done. I don't think I missed any, but if you see one let me know